packages/p2p/src/constants/modals.ts

```diff
@@ -0,0 +1,163 @@
+            )
+    ),
+    ErrorModal: React.lazy(
+        () => import(/* webpackChunkName: "error-modal" */ 'Components/modal-manager/modals/error-modal')
+    ),
+    FilterModal: React.lazy(
+        () => import(/* webpackChunkName: "filter-modal" */ 'Components/modal-manager/modals/filter-modal')
+    ),
+    InvalidVerificationLinkModal: React.lazy(
+        () =>
+            import(
+                /* webpackChunkName: "invalid-verification-link-modal" */ 'Components/modal-manager/modals/invalid-verification-link-modal'
+            )
+    ),
+    LoadingModal: React.lazy(
+        () => import(/* webpackChunkName: "loading-modal" */ 'Components/modal-manager/modals/loading-modal')
+    ),
+    MarketRateChangeErrorModal: React.lazy(
+        () =>
+            import(
+                /* webpackChunkName: "market-rate-change-error-modal" */ 'Components/modal-manager/modals/market-rate-change-error-modal'
+            )
+    ),
+    MyAdsDeleteModal: React.lazy(
+        () =>
+            import(/* webpackChunkName: "my-ads-delete-modal" */ 'Components/modal-manager/modals/my-ads-delete-modal')
+    ),
+    MyAdsDeleteErrorModal: React.lazy(
+        () =>
+            import(
+                /* webpackChunkName: "my-ads-delete-error-modal" */ 'Components/modal-manager/modals/my-ads-delete-error-modal'
+            )
+    ),
+    MyAdsFloatingRateSwitchModal: React.lazy(
+        () =>
+            import(
+                /* webpackChunkName: "my-ads-floating-rate-switch-modal" */ 'Components/modal-manager/modals/my-ads-floating-rate-switch-modal'
+            )
+    ),
+    OrderDetailsCancelModal: React.lazy(
+        () =>
+            import(
+                /* webpackChunkName: "order-details-cancel-modal" */ 'Components/modal-manager/modals/order-details-cancel-modal'
+            )
+    ),
+    OrderDetailsConfirmModal: React.lazy(
+        () =>
+            import(
+                /* webpackChunkName: "order-details-confirm-modal" */ 'Components/modal-manager/modals/order-details-confirm-modal'
+            )
+    ),
+    QuickAddModal: React.lazy(
+        () => import(/* webpackChunkName: "quick-add-modal" */ 'Components/modal-manager/modals/quick-add-modal')
+    ),
+    RatingModal: React.lazy(
+        () => import(/* webpackChunkName: "rating-modal" */ 'Components/modal-manager/modals/rating-modal')
+    ),
+    RateChangeModal: React.lazy(
+        () => import(/* webpackChunkName: "rate-change-modal" */ 'Components/modal-manager/modals/rate-change-modal')
+    ),
+    RecommendedModal: React.lazy(
+        () => import(/* webpackChunkName: "recommended-modal" */ 'Components/modal-manager/modals/recommended-modal')
+    ),
+} as const;
+
+export type TModals = typeof Modals;
+export type TModalKeys = keyof TModals;
+export type TModalProps = {
+    [T in TModalKeys]: Parameters<TModals[T]>[0] extends { [key: string]: any }
+        ? Parameters<TModals[T]>[0]
+        : Record<string, never>;
```

Comment:
Couldn't use a default `{}` to represent modals without props as eslint was preferring to use `Record<string, never>`:
![Screenshot 2023-07-07 at 2 32 43 PM](https://github.com/farrah-deriv/deriv-app/assets/103016120/583c3c8b-09d4-4937-a35b-dbb67d3bbe20)